### DISCUSSION TITLE
How to pass custom parameter to thin clients

### DISCUSSION THREAD

**Discussion by hc2021hc** [ltsp/ltsp#483] [General]
I have an application where the LTSP thin clients running Ubuntu 20.04 will each use a unique parameter value for the application.  May I know if we can specify such a parameter in the LTSP server and then each LTSP client can read the unique value, say, from a config file, for the application? Or we can simply use ltsp.conf to convey the parameter?

**Comment by alkisg** [comment]
One way to do it is to put the parameter in a file:

```shell
[mac:01]
POST_INIT_SET_PARAM="echo PARAMETER=01 > /etc/ltsp/application.conf"

[mac:02]
POST_INIT_SET_PARAM="echo PARAMETER=02 > /etc/ltsp/application.conf"
```

An alternative way to do it is to modify an existing text file (e.g. your shell script launcher) with sed:

```shell
[mac:01]
POST_INIT_SET_PARAM="sed 's/param=temp/param=01/' -i /usr/bin/launcher.sh"

[mac:02]
POST_INIT_SET_PARAM="sed 's/param=temp/param=02/' -i /usr/bin/launcher.sh"
```

**Comment by hc2021hc** [comment]
Thanks.  I use the first method and it works perfectly!
